jormungandr/src/settings/start/trusted_peer.rs

```diff
@@ -0,0 +1,146 @@
+use crate::{network::p2p::Id, settings::start::config::Address};
+use multiaddr::AddrComponent;
+use serde::{de::Visitor, Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    fmt,
+    iter::FromIterator,
+    net::{SocketAddr, ToSocketAddrs},
+};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct TrustedPeer {
+    pub address: TrustedAddress,
+    pub id: Id,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TrustedAddress(pub multiaddr::Multiaddr);
+
+custom_error! {pub AddressError
+    DnsLookupError { source: std::io::Error } = "failed to resolve DNS name {source}",
+    NoPortSpecified = "no TCP port specified",
+    NoAppropriateDNSFound = "the address was resolved, but it doesn't provide IPv4 or IPv6 addresses",
+    UnsupportedProtocol = "the provided protocol is unsupported, please use one of ip4/ip6/dns4/dns6",
+}
+
+impl TrustedAddress {
+    pub fn to_addresses(&self) -> Result<Vec<Address>, AddressError> {
```

Comment:
Please add a prominent (doc) comment notice that a blocking network resolver roundtrip happens here, so that we remember to avoid calling this in async code.